### ISSUE TITLE
[Bug]: Open PR not updated if parameter changed

### ISSUE THREAD

**Issue by kdeldycke** [AndreasAugustin/actions-template-sync#481]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Following-up on #437, `actions-template-sync` makes sure one and one only PR is kept open. That way, there is only one PR open even if `actions-template-sync` is run multiple times.

But if we change metadata, like the PR title, between workflow runs, the PR is not updated. That's because `actions-template-sync` only decides to update a PR if and only if files have been changed between workflow runs.

### Expected Behavior

I expect the open PR to be kept in sync with its new parameters even if no changes are noticed by `actions-template-sync` in the commits and their content.

### Current Behavior

If you have a standard step like:
```yaml
      - name: actions-template-sync
        uses: AndreasAugustin/actions-template-sync@v1
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          source_repo_path: kdeldycke/awesome-template
          pr_title: "[sync] Updates from `awesome-template` repository"
          pr_commit_msg: "[sync] Updates from `awesome-template` repository"
          pr_labels: "📚 documentation"
```

Then a PR is created and left open by `actions-template-sync`.

Now keep this PR open, and then immediately change the `pr_title` to something else like:
```yaml
      - name: actions-template-sync
        uses: AndreasAugustin/actions-template-sync@v1
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          source_repo_path: kdeldycke/awesome-template
          pr_title: "Brand new PR title!"
          pr_commit_msg: "[sync] Updates from `awesome-template` repository"
          pr_labels: "📚 documentation"
```

A workflow run is triggered, but the PR is not updated with the new title.

### Possible Solution

`actions-template-sync` should compare the content of the PR, including its data and metadata to decide if that PR should be updated or not.

### Template sync version Version

1.8.1

**Comment by AndreasAugustin** [comment]
Hi @kdeldycke . Sorry for the delay.
Decided to have the option to opt in to force the update (#482 ) https://github.com/AndreasAugustin/actions-template-sync/issues/482#issuecomment-1995822745

This bug will be closed with #499

**Comment by kdeldycke** [comment]
I just tested this and indeed, `is_force_push_pr: true` let the PR be updated even if only the PR metadata are changed. Thanks @AndreasAugustin for the fix!

Now from a usability point of view, and to make things more explicit, I would suggest to rename this parameter to `only_update_on_file_changes` or something similar, and reverse the boolean logic. But that's a matter of preferences and is up to you.
